### ISSUE TITLE
Example repository

### ISSUE THREAD

**Issue by LukeMathWalker** [rust-ndarray/ndarray#674]
In https://github.com/rust-ndarray/ndarray-linalg/pull/166 @munckymagik advanced the idea of adding a new repository to `rust-ndarray` to host more complex standalone examples.

It could be a good place to showcase what is possible using the different pieces of the ecosystem, in a sort of cookbook format.

I love the idea and I think it could provide a lot of value. What are your thoughts?

**Comment by max-sixty** [comment]
Yes agree! Personally I'd also find a side-by-side example of numpy / scipy v helpful, furthering the existing docs there. Even without that, I think it's a great idea.

**Comment by jturner314** [comment]
That sounds like a good idea to me. We do have a few examples in modules linked at the bottom of [`ndarray_for_numpy_users`](https://docs.rs/ndarray/0.12.1/ndarray/doc/ndarray_for_numpy_users/index.html), but they're pretty minimal and focus more on `ndarray` than the ecosystem as a whole. We should also review the files in the [`examples` directory](https://github.com/rust-ndarray/ndarray/tree/master/examples) to see if we need to update/remove any of them.

**Comment by LukeMathWalker** [comment]
I have been going through the files in the examples directory and I have to admit that it's not obvious what we are trying to show in some of them.
When I'll be back I'll do a second pass and ask for help on those that remain obscure.

On 25 Jul 2019, at 23:11, Jim Turner <notifications@github.com<mailto:notifications@github.com>> wrote:

That sounds like a good idea to me. We do have a few examples in modules linked at the bottom of ndarray_for_numpy_users<https://docs.rs/ndarray/0.12.1/ndarray/doc/ndarray_for_numpy_users/index.html>, but they're pretty minimal and focus more on ndarray than the ecosystem as a whole. We should also review the files in the examples directory<https://github.com/rust-ndarray/ndarray/tree/master/examples> to see if we need to update/remove any of them.

**Comment by LukeMathWalker** [comment]
First sketch is out: https://github.com/rust-ndarray/ndarray-examples

I still need to add a CI pipeline (mirroring @termoshtt's work on `ndarray-linalg`) - suggestions are welcome :+1:

**Comment by LukeMathWalker** [comment]
Backend selection is now working as expected. Unfortunately, it seems we can't provide a top-level backend selection mechanism due to https://github.com/rust-lang/cargo/issues/5015 - it has to be done in the `Cargo.toml` of each example.

**Comment by LukeMathWalker** [comment]
I have another PR up, with a simple implementation of K-Means clustering: https://github.com/rust-ndarray/ndarray-examples/pull/2

Looking for :eyes:
